crates/astria-sequencer-relayer/src/relayer.rs

```diff
@@ -84,75 +98,244 @@ impl Relayer {
-            return Ok(new_state);
+            Err(e) => report_err!(e, "failed getting latest block from sequencer"),
         }
+    }
 
-        let sequencer_block = match SequencerBlockData::from_tendermint_block(resp.block) {
-            Ok(block) => block,
+    /// Handle the result
+    #[instrument(skip_all)]
+    fn handle_conversion_completed(
+        &mut self,
+        join_result: Result<eyre::Result<Option<SequencerBlockData>>, task::JoinError>,
+    ) -> HandleConversionCompletedResult {
+        // First check if the join task panicked
+        let conversion_result = match join_result {
+            Ok(conversion_result) => conversion_result,
+            // Report if the task failed, i.e. panicked
             Err(e) => {
-                warn!(error = ?e, "failed to convert block to DA block");
-                return Ok(new_state);
+                // TODO: inject the correct tracing span
+                report_err!(e, "conversion task failed");
+                return HandleConversionCompletedResult::Handled;
             }
         };
-
-        self.block_tx.send(sequencer_block.clone())?;
-        let namespace_count = sequencer_block.rollup_txs.len();
-        if let Some(client) = &self.data_availability_client {
-            match client
-                .submit_block(
-                    sequencer_block,
-                    &self.validator.signing_key,
-                    self.validator.verification_key,
-                )
-                .await
-            {
-                Ok(resp) => {
-                    new_state
-                        .current_data_availability_height
-                        .replace(resp.height);
-                    info!(
-                        sequencer_block = height,
-                        da_layer_block = resp.height,
-                        namespace_count = namespace_count,
-                        "submitted sequencer block to DA layer",
-                    );
+        // Then handle the actual result of the computation
+        match conversion_result {
+            // Gossip and collect successfully converted sequencer responses
+            Ok(Some(sequencer_block_data)) => {
+                if self
+                    .gossip_block_tx
+                    .send(sequencer_block_data.clone())
+                    .is_err()
+                {
+                    return HandleConversionCompletedResult::GossipChannelClosed;
                 }
-                Err(e) => warn!(error = ?e, "failed to submit block to DA layer"),
+                // Update the internal state if the block was admitted
+                let height = sequencer_block_data.header.height.value();
+                self.state_tx.send_if_modified(|state| {
+                    if Some(height) > state.current_sequencer_height {
+                        state.current_sequencer_height = Some(height);
+                        return true;
+                    }
+                    false
```

Comment:
is there ever a case where the height wouldn't be higher than the current? it seems guaranteed by `convert_block_response_to_sequencer_block_data`?

Reply:
Task conversion happens in a joinset. Albeit unlikely, you are not guaranteed that the conversions finish in the order of their heights. For example, if 101 is very big and takes a long time to convert, but block 102 is very small and conversion finishes immediately, then you don't want to update the internal state after 101 is done converting.

On that note: what happens if we bundle sequencer blocks 100 and 102 and submit them to celestia, but 101 only finishes after. Is it ok to submit it out of order? Or should we ensure that there is never a gap?

Reply:
i think submitting out of order is ok - the conductor should be able to handle it eventually (see #15)

Reply:
I believe that right now submitting out of order does not work. 

This should be logic we have in conductor, but could break functionality until then.